proposals/stf-general-investment.md

```diff
@@ -0,0 +1,267 @@
+Apart from implementing technical measures to that end, merely having qualified engineers available for ongoing maintenance and addressing small-scale, long-term, or cross-cutting issues as part of their engagement will improve the overall health of the ecosystem by unblocking volunteer efforts more quickly than is currently possible.
+We expect this to have a compounding effect on overall productivity, as we estimate that more than 95% of the work is currently done by volunteers, many of whom are slowed down by lack of support and swift decision making.
+
+- Improve the contributor experience (300k)
+
+  Streamlining the contribution process should make it more attractive for newcomers to participate in development, and keep existing people around and motivated.
+  Growing the contributor community will make the ecosystem more robust against fluctuation.
+  We want to achieve this by making more systemic knowledge explicit and easier to discover, access, and manipulate, and thus ease onboarding of new contributors, and reduce contribution turnaround time.
+
+  - Tools for review automation (50k)
+  - Improve code organisation in C++ Nix (30k)
+  - Increase coverage and reduce duration of Nix integration testing (30k)
+  - Monitoring and data gathering for informed decision-making (30k)
+  - Professionalise the moderation team (50k)
+  - Organisation-wide declarative permission and notification management (60k)
+
+- Improve documentation (170k)
+
+  More focused, correct, up-to-date, and well-structured documentation will make the system more learnable.
+  This will over time reduce the number and size of blind spots in components that may otherwise end up unmaintained, a problem we are observing for many years now.
+  It should also tangibly reduce maintainer workload by making users and contributors more self-sufficient.
+
+  - Improve and speed up documentation rendering infrastructure (40k)
+  - Rework the Nixpkgs manual (60k)
+  - [Design and implement an information architecture](https://discourse.nixos.org/t/zurich-23-11-zhf-hackathon-and-ux-workshop-report/37848#ux-workshop-5) (50k)
+  - Build documentation testing infrastructure (20k)
+
+- Reduce ongoing costs for continuous integration and distribution (180k)
+
+  Infrastructure expenses are a significant risk for the ecosystem, because they are almost completely sponsored by companies who may end their support any time.
+  Reducing these ongoing costs will make it easier to migrate between providers or even self-host.
+  Improving CI performance will also contribute to faster workflows, and increase responsiveness to security incidents.
+  This will also impact all downstream users and significantly reduce their storage and energy requirements for the forseeable future.
+
+  - Deduplicate the cache storage (40k)
+  - Improve evaluation performance (100k)
+    - parallel evaluation
```

Comment:
Another option @RaitoBezarius reminded me of is the bytecode VM approach Tvix took. Probably we should rephrase this objective as "investigate and implement a substantial reduction of computational requirements for evaluating NixOS configurations" to make it meaningfully open-ended but targeted at actually reducing the resource consumption.